ksqldb-execution/src/main/java/io/confluent/ksql/execution/interpreter/terms/LambdaFunctionTerms.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.interpreter.terms;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import io.confluent.ksql.execution.codegen.helpers.TriFunction;
+import io.confluent.ksql.execution.interpreter.TermEvaluationContext;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.util.Pair;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public class LambdaFunctionTerms {
+
+  public static class LambdaFunction1Term implements Term {
+
+    private final List<Pair<String, SqlType>> argNamesToTypes;
+    private final Term body;
+
+    public LambdaFunction1Term(final List<Pair<String, SqlType>> argNamesToTypes, final Term body) {
```

Comment:
Could we enforce the `body` to be `FunctionCallTerm`?

Reply:
The body is whatever is actually in the body of the lambda, which can be anything (e.g. `X+1` of the lambda definition `X => X+1`).  I think you're meaning the parent of this term, which is always a function call.  We don't actually keep parent references, and the children of a function call can be anything, so I don't think we can enforce types here.